### PR TITLE
clipp: modernize

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **clipp/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

I was looking for an up-to-date minimal example but I did not find anything so I pushed this one a little. farther 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/clipp/all/conanfile.py [hunk 1]
```diff
@@ -17,14 +18,12 @@ def _source_subfolder(self):
         return "source_subfolder"
```
Comment (uilianries) [inline recipes/clipp/all/conanfile.py:18]:
```suggestion
        return os.path.join(self.source_folder, "source_subfolder")
```

File: recipes/clipp/all/conanfile.py [hunk 2]
```diff
@@ -17,14 +18,12 @@ def _source_subfolder(self):
...
-        extracted_dir = glob.glob(self.name + "-*/")[0]
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.files.get(self, **self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def package(self):
-        include_folder = os.path.join(self._source_subfolder, "include")
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="*", dst="include", src=include_folder)
+        tools.files.copy(self, "LICENSE", self._source_subfolder, "licenses")
```
Comment (uilianries) [inline recipes/clipp/all/conanfile.py:25]:
```suggestion
        tools.files.copy(self, "LICENSE", src=self._source_subfolder, dst=os.path.join(self.package_folder, "licenses"))
```

File: recipes/clipp/all/conanfile.py [hunk 3]
```diff
@@ -17,14 +18,12 @@ def _source_subfolder(self):
...
+        tools.files.copy(self, "*", os.path.join(self._source_subfolder, "include"), "include")
```
Comment (uilianries) [inline recipes/clipp/all/conanfile.py:26]:
```suggestion
        tools.files.copy(self, "*", src=os.path.join(self._source_subfolder, "include"), dst=os.path.join(self.package_folder, "include"))
```

File: recipes/clipp/all/conanfile.py [hunk 4]
```diff
@@ -17,14 +18,12 @@ def _source_subfolder(self):
...
 
     def package_id(self):
         self.info.header_only()
```
Comment (uilianries) [inline recipes/clipp/all/conanfile.py:29]:
```suggestion
        self.info.clear()
```

File: recipes/clipp/all/conanfile.py [hunk 5]
```diff
@@ -1,30 +1,25 @@
 import os
-import glob
 from conan import ConanFile, tools
 
 
+required_conan_version = ">=1.46.0"
```
Comment (uilianries) [inline recipes/clipp/all/conanfile.py:5]:
```suggestion
required_conan_version = ">=1.50.0"
```

`self.info.clear()`  https://docs.conan.io/en/latest/changelog.html?highlight=clear#jun-2022

**Review by uilianries — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Pull request merged**